OrbitGl/TrackManager.h

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#pragma once
+
+#include <stdint.h>  // for int32_t, uint64_t, uint32_t
+#include <stdlib.h>  // for abs
+
+#include <map>     // for map
+#include <memory>  // for shared_ptr
+#include <mutex>   // for recursive_mutex
+#include <string>  // for string
+#include <unordered_map>
+#include <vector>  // for vector
+
+#include "AsyncTrack.h"
+#include "FrameTrack.h"
+#include "GpuTrack.h"
+#include "GraphTrack.h"
+#include "PickingManager.h"  // for PickingMode
+#include "SchedulerTrack.h"
+#include "StringManager.h"  // for StringManager
+#include "ThreadTrack.h"
+#include "Timer.h"            // for Timer
+#include "Track.h"            // for Track
+#include "capture_data.pb.h"  // for FunctionInfo
+
+class TimeGraph;
+
+class TrackManager {
+ public:
+  explicit TrackManager(TimeGraph* time_graph);
+
+  void Clear();
+
+  [[nodiscard]] std::vector<Track*> GetTracks() const;
+  [[nodiscard]] std::vector<Track*> GetFilteredTracks() const { return sorted_filtered_tracks_; }
+  [[nodiscard]] std::vector<ThreadTrack*> GetThreadTracks() const;
+  [[nodiscard]] std::vector<FrameTrack*> GetFrameTracks() const;
+
+  [[nodiscard]] ThreadTrack* GetTracepointsSystemWideTrack() const {
+    return tracepoints_system_wide_track_;
+  }
+
+  void SetStringManager(std::shared_ptr<StringManager> str_manager);
+  [[nodiscard]] StringManager* GetStringManager() const { return string_manager_.get(); }
+
+  void SortTracks();
+  void SetFilter(const std::string& filter);
+  void UpdateFilteredTrackList();
+
+  void UpdateTracks(uint64_t min_tick, uint64_t max_tick, PickingMode picking_mode);
+  float GetTracksHeight() const { return std::abs(tracks_height_); }
+
+  SchedulerTrack* GetOrCreateSchedulerTrack();
```

Comment:
Can those be `const ...Track*(...) const;`? I assume that not, but maybe we should work toward this.
Also as soon as we have create and get seperated, we should have the getters be nodisard.

Reply:
Something that with Henning we discussed a little about. For now, I'm leaving as it is, but definitely I want to split it. In fact, I want to have an only list of Tracks and only 1 function to get Tracks (with a template?), but not sure if I will succeed with that.

Will add a TODO here.

Reply:
Sounds good, it's perfectly fine to address that later.